src/map/clif.c

```diff
@@ -9951,8 +9951,13 @@ static void clif_mobname_additional_ack(int fd, struct block_list *bl)
 		str_p += sprintf(str_p, "HP: %u%% | ", get_percentage(md->status.hp, md->status.max_hp));
 	//Even thought mobhp ain't a name, we send it as one so the client
 	//can parse it. [Skotlex]
-	if (str_p != mobhp) {
-		*(str_p-3) = '\0'; //Remove trailing space + pipe.
+	if ((battle_config.show_mob_info & (4 + 1 + 2)) != 0 && str_p != mobhp) {
```

Comment:
The result is the same, but I think this is more error-proof if/when adding new values in the future:

```suggestion
	if ((battle_config.show_mob_info & (4 | 1 | 2)) != 0 && str_p != mobhp) {
```